Fix an error with block-scoped-var when using ecmaFeature modules

Cellule opened the pull request:
Fixes https://github.com/eslint/eslint/issues/2242

michaelficarra commented:
This will cause problems with FunctionExpression scoping. They should not be implemented with the same handler. Add a test for `!function f(){}; f`.

Cellule commented:
You were right. I also added a few more tests to try and cover everything

michaelficarra commented:
LGTM.

Cellule commented:
I just realized that there seems to be the same kind of problem with classes. Do you want me to fix it in this pr or do another one ?

michaelficarra commented:
You should probably open a separate issue and PR.

michaelficarra commented:
Also, read the contribution guide. The commit messages have to have a certain format.

Cellule commented:
Ok, I'll check tomorrow for the other issue.
I hope this format is good for my commit.

nzakas commented:
You need to mention the issue number in the commit message. 

Also, please sign our CLA: http://eslint.org/cla

nzakas commented on lib/rules/block-scoped-var.js:
```diff
@@ -173,11 +173,20 @@ module.exports = function(context) {
         declare(["arguments"]);
     }
 
+    function functionDeclarationHandler(node) {
```
Please add JSDoc comments to these functions. They are so similar that it would be hard for most to understand the difference.

Cellule commented:
- I fixed the commit message to include the issue and not the PR.
- I added JSDoc to my new function and also the one I modified.
- I signed the cla.

nzakas commented:
Looks good. Thanks!
